internal/ocm/resource/dns_domain.go

```diff
@@ -0,0 +1,39 @@
+package resource
+
+import (
+	cmv1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
+)
+
+type DNSDomain struct {
+	client  *cmv1.DNSDomainsClient
+	builder *cmv1.DNSDomainBuilder
+	id      string
+}
+
+func NewDNSDomain(client *cmv1.DNSDomainsClient) *DNSDomain {
+	return &DNSDomain{
+		client:  client,
+		builder: cmv1.NewDNSDomain(),
+	}
+}
+
+func (d *DNSDomain) GetDNSDomainBuilder() *cmv1.DNSDomainBuilder {
+	return d.builder
+}
+
+func (d *DNSDomain) Create() (*cmv1.DNSDomainsAddResponse, error) {
+	payload, err := d.builder.Build()
+	if err != nil {
+		return nil, err
+	}
+	return d.client.Add().Body(payload).Send()
+}
+
+func (d *DNSDomain) Delete(id string) error {
+	_, err := d.client.DNSDomain(id).Delete().Send()
```

Comment:
Is it the right convention to have the client inside the `internal/ocm/resource`?
This is not how it was done in the cluster
Lets discuss about what we think should be the right convention

Reply:
It's not the same, It's simpler. Less logic in this resource. Don't see the benefit of exposing the `builder` in this case. Provider in this case simply needs some functions for CRUD. Passing minimal information (mainly id). No reason to add more lines of code for these small operations. IMO

But sure, We can talk about it :)